app/src/main/kotlin/com/intive/picover/auth/repository/AuthRepository.kt

```diff
@@ -41,15 +43,14 @@ class AuthRepository @Inject constructor(
 		firebaseAuth.signOut()
 	}
 
-	suspend fun userProfile(): Profile {
-		val photoUrl = userAvatarReference.downloadUrl.await()
-		return requireUser().let {
-			Profile(
-				photo = photoUrl,
-				name = it.displayName!!,
-				email = it.email!!,
-			)
-		}
+	suspend fun fetchUserProfile(): Result<Profile> = suspendCoroutine { continuation ->
```

Comment:
Is is needed here to specify the return type?

Reply:
Done